### PR TITLE
Dumped blobs validation

### PR DESCRIPTION
**Pull request by iam-theKid**
Validating dumped blobs against OTA and IPSW to confirm it can be used to restore same iOS version in case of disaster.

### CODE REVIEW

**Comment by itsnebulalol** [comment]
Can you make it do it automatically? Use the $version variable.

After this PR is finished, can you do the same for the tweaks branch

**Comment by iam-theKid** [comment]
Done.

File: palera1n.sh [hunk 1]
```diff
@@ -309,6 +309,40 @@ if [ ! -f blobs/"$deviceid"-"$version".shsh2 ]; then
...
+        echo "\033[0m"
+    elif [[ "$($oscheck/img4tool --verify BuildManifest\_IPSW\_$deviceid\_$blobsVersion\_$blobsBuildID.plist -s dumped.shsh | grep 'APTicket' | cut -d ' ' -f4)" == "GOOD\!" ]]; then
+        echo "\033[32m"
+        "$dir"/img4tool --verify work/BuildManifest\_IPSW\_$deviceid\_$blobsVersion\_$blobsBuildID.plist -s $dumpedBlobs | grep 'APTicket'
+        "$dir"/img4tool --verify work/BuildManifest\_IPSW\_$deviceid\_$blobsVersion\_$blobsBuildID.plist -s $dumpedBlobs | grep 'IM4M is valid\|BuildNumber\|\|DeiviceClass\|RestoreBehavior\|Variant\|SHSH2'
+        echo "\033[0m "
+    else
+        echo "\033[91m[IMG4TOOL] APTicket is BAD!\033[0m"
+    fi
+    read -p "Press any key to continue..."
```
Comment (itsnebulalol) [inline palera1n.sh:344]: Can you remove this too please

File: palera1n.sh [hunk 2]
```diff
@@ -309,6 +309,40 @@ if [ ! -f blobs/"$deviceid"-"$version".shsh2 ]; then
...
+        "$dir"/img4tool --verify work/BuildManifest\_OTA\_$deviceid\_$blobsVersion\_$blobsBuildID.plist -s $dumpedBlobs | grep 'APTicket'
+        "$dir"/img4tool --verify work/BuildManifest\_OTA\_$deviceid\_$blobsVersion\_$blobsBuildID.plist -s $dumpedBlobs | grep 'IM4M is valid\|BuildNumber\|\|DeiviceClass\|RestoreBehavior\|Variant\|SHSH2'
+        echo "\033[0m"
+    elif [[ "$($oscheck/img4tool --verify BuildManifest\_IPSW\_$deviceid\_$blobsVersion\_$blobsBuildID.plist -s dumped.shsh | grep 'APTicket' | cut -d ' ' -f4)" == "GOOD\!" ]]; then
+        echo "\033[32m"
+        "$dir"/img4tool --verify work/BuildManifest\_IPSW\_$deviceid\_$blobsVersion\_$blobsBuildID.plist -s $dumpedBlobs | grep 'APTicket'
+        "$dir"/img4tool --verify work/BuildManifest\_IPSW\_$deviceid\_$blobsVersion\_$blobsBuildID.plist -s $dumpedBlobs | grep 'IM4M is valid\|BuildNumber\|\|DeiviceClass\|RestoreBehavior\|Variant\|SHSH2'
+        echo "\033[0m "
+    else
+        echo "\033[91m[IMG4TOOL] APTicket is BAD!\033[0m"
```
Comment (itsnebulalol) [inline palera1n.sh:342]: Make these echos output to the debug log, not normal log

File: palera1n.sh (hunk 1 above)
Comment (iam-theKid) [reply palera1n.sh:344]: what should happen if blobs are not valid? pls suggest.

**Comment by iam-theKid** [comment]
Whwen you have some time, let me know how it looks now

**Comment by itsnebulalol** [comment]
Manually merged in tweaks branch, not going to do it in rootless for now (will do it eventually)

**Pull request closed without merging**